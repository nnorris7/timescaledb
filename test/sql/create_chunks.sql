\ir include/create_single_db.sql

CREATE TABLE chunk_test(
        time       BIGINT,
        metric     INTEGER,
        device_id  TEXT
    );

-- Test chunk closing/creation
SELECT * FROM create_hypertable('chunk_test', 'time', 'device_id', 2, chunk_time_interval => 10);
SELECT * FROM _timescaledb_catalog.hypertable;

INSERT INTO chunk_test VALUES (1, 1, 'dev1'),
                              (2, 2, 'dev2'),
                              (45, 2, 'dev2'),
                              (46, 2, 'dev2');

SELECT * FROM set_chunk_time_interval('chunk_test', 40::bigint);


INSERT INTO chunk_test VALUES(23, 3, 'dev3');

SELECT * FROM chunk_test order by time, metric, device_id;
SELECT * FROM _timescaledb_catalog.chunk;
SELECT * FROM _timescaledb_catalog.hypertable;

SELECT * FROM ONLY chunk_test;
SELECT * FROM _timescaledb_catalog.chunk c
    LEFT JOIN _timescaledb_catalog.chunk_constraint cc ON (c.id = cc.chunk_id)
    LEFT JOIN _timescaledb_catalog.dimension_slice ds ON (ds.id = cc.dimension_slice_id)
    LEFT JOIN _timescaledb_catalog.dimension d ON (d.id = ds.dimension_id)
    LEFT JOIN _timescaledb_catalog.hypertable h ON (d.hypertable_id = h.id)
    WHERE h.schema_name = 'public' AND h.table_name = 'chunk_test'
    ORDER BY c.id, d.id;

