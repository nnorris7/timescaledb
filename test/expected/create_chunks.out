\ir include/create_single_db.sql
SET client_min_messages = WARNING;
DROP DATABASE IF EXISTS single;
SET client_min_messages = NOTICE;
CREATE DATABASE single;
\c single
CREATE EXTENSION IF NOT EXISTS timescaledb;
CREATE TABLE chunk_test(
        time       BIGINT,
        metric     INTEGER,
        device_id  TEXT
    );
-- Test chunk closing/creation
SELECT * FROM create_hypertable('chunk_test', 'time', 'device_id', 2, chunk_time_interval => 10);
 create_hypertable 
-------------------
 
(1 row)

SELECT * FROM _timescaledb_catalog.hypertable;
 id | schema_name | table_name | associated_schema_name | associated_table_prefix | num_dimensions 
----+-------------+------------+------------------------+-------------------------+----------------
  1 | public      | chunk_test | _timescaledb_internal  | _hyper_1                |              2
(1 row)

INSERT INTO chunk_test VALUES (1, 1, 'dev1'),
                              (2, 2, 'dev2'),
                              (45, 2, 'dev2'),
                              (46, 2, 'dev2');
SELECT * FROM set_chunk_time_interval('chunk_test', 40::bigint);
 set_chunk_time_interval 
-------------------------
 
(1 row)

INSERT INTO chunk_test VALUES(23, 3, 'dev3');
SELECT * FROM chunk_test order by time, metric, device_id;
 time | metric | device_id 
------+--------+-----------
    1 |      1 | dev1
    2 |      2 | dev2
   23 |      3 | dev3
   45 |      2 | dev2
   46 |      2 | dev2
(5 rows)

SELECT * FROM _timescaledb_catalog.chunk;
 id | hypertable_id |      schema_name      |    table_name    
----+---------------+-----------------------+------------------
  1 |             1 | _timescaledb_internal | _hyper_1_1_chunk
  2 |             1 | _timescaledb_internal | _hyper_1_2_chunk
  3 |             1 | _timescaledb_internal | _hyper_1_3_chunk
  4 |             1 | _timescaledb_internal | _hyper_1_4_chunk
(4 rows)

SELECT * FROM _timescaledb_catalog.hypertable;
 id | schema_name | table_name | associated_schema_name | associated_table_prefix | num_dimensions 
----+-------------+------------+------------------------+-------------------------+----------------
  1 | public      | chunk_test | _timescaledb_internal  | _hyper_1                |              2
(1 row)

SELECT * FROM ONLY chunk_test;
 time | metric | device_id 
------+--------+-----------
(0 rows)

SELECT * FROM _timescaledb_catalog.chunk c
    LEFT JOIN _timescaledb_catalog.chunk_constraint cc ON (c.id = cc.chunk_id)
    LEFT JOIN _timescaledb_catalog.dimension_slice ds ON (ds.id = cc.dimension_slice_id)
    LEFT JOIN _timescaledb_catalog.dimension d ON (d.id = ds.dimension_id)
    LEFT JOIN _timescaledb_catalog.hypertable h ON (d.hypertable_id = h.id)
    WHERE h.schema_name = 'public' AND h.table_name = 'chunk_test'
    ORDER BY c.id, d.id;
 id | hypertable_id |      schema_name      |    table_name    | chunk_id | dimension_slice_id | id | dimension_id | range_start | range_end  | id | hypertable_id | column_name | column_type | num_slices | partitioning_func_schema |   partitioning_func   | interval_length | id | schema_name | table_name | associated_schema_name | associated_table_prefix | num_dimensions 
----+---------------+-----------------------+------------------+----------+--------------------+----+--------------+-------------+------------+----+---------------+-------------+-------------+------------+--------------------------+-----------------------+-----------------+----+-------------+------------+------------------------+-------------------------+----------------
  1 |             1 | _timescaledb_internal | _hyper_1_1_chunk |        1 |                  1 |  1 |            1 |           0 |         10 |  1 |             1 | time        | bigint      |            |                          |                       |              40 |  1 | public      | chunk_test | _timescaledb_internal  | _hyper_1                |              2
  1 |             1 | _timescaledb_internal | _hyper_1_1_chunk |        1 |                  2 |  2 |            2 |  1073741823 | 2147483647 |  2 |             1 | device_id   | text        |          2 | _timescaledb_internal    | get_partition_for_key |                 |  1 | public      | chunk_test | _timescaledb_internal  | _hyper_1                |              2
  2 |             1 | _timescaledb_internal | _hyper_1_2_chunk |        2 |                  1 |  1 |            1 |           0 |         10 |  1 |             1 | time        | bigint      |            |                          |                       |              40 |  1 | public      | chunk_test | _timescaledb_internal  | _hyper_1                |              2
  2 |             1 | _timescaledb_internal | _hyper_1_2_chunk |        2 |                  4 |  4 |            2 |           0 | 1073741823 |  2 |             1 | device_id   | text        |          2 | _timescaledb_internal    | get_partition_for_key |                 |  1 | public      | chunk_test | _timescaledb_internal  | _hyper_1                |              2
  3 |             1 | _timescaledb_internal | _hyper_1_3_chunk |        3 |                  5 |  5 |            1 |          40 |         50 |  1 |             1 | time        | bigint      |            |                          |                       |              40 |  1 | public      | chunk_test | _timescaledb_internal  | _hyper_1                |              2
  3 |             1 | _timescaledb_internal | _hyper_1_3_chunk |        3 |                  4 |  4 |            2 |           0 | 1073741823 |  2 |             1 | device_id   | text        |          2 | _timescaledb_internal    | get_partition_for_key |                 |  1 | public      | chunk_test | _timescaledb_internal  | _hyper_1                |              2
  4 |             1 | _timescaledb_internal | _hyper_1_4_chunk |        4 |                  7 |  7 |            1 |          10 |         40 |  1 |             1 | time        | bigint      |            |                          |                       |              40 |  1 | public      | chunk_test | _timescaledb_internal  | _hyper_1                |              2
  4 |             1 | _timescaledb_internal | _hyper_1_4_chunk |        4 |                  4 |  4 |            2 |           0 | 1073741823 |  2 |             1 | device_id   | text        |          2 | _timescaledb_internal    | get_partition_for_key |                 |  1 | public      | chunk_test | _timescaledb_internal  | _hyper_1                |              2
(8 rows)

